mmdet/datasets/dataset_wrappers.py

```diff
@@ -280,3 +284,343 @@ def __getitem__(self, idx):
+    - mosaic and mixup transform will be disabled last several epochs
+      controlled by YoloXProcessHook
+    - mosaic image will be futher augment by mosaic pipeline(default RandomAffine)
+    - mosaic and mixup image will be futher augment by pipeline
+    Also MosaicMixUpDataset will only be used in train mode
+
+    The logic of mixup transform:
+
+                        mixup transform
+               +------------------------------+
+               | mixup image   |              |
+               |      +--------|--------+     |
+               |      |        |        |     |
+               |---------------+        |     |
+               |      |  Affined mosaic |     |
+               |      |      image      |     |
+               |      |                 |     |
+               |      |                 |     |
+               |      |-----------------+     |
+               |             pad              |
+               +------------------------------+
+    The mixup transform step:
+    - Another random image is picked by dataset  and embedded in the top left patch(after padding and resizing)
+    - The target of mixup transform is the weighted average of mixup image and mosaic image.
+
+
+    dataset: Custom dataset with LoadImageFromFile and LoadAnnotations pipeline
+    mosaic_pipeline: default RandomAffine, applied after mosaic before mixup
+    pipeline: custom pipeline after mosaic and mixup transform
+
+   Args:
+       dataset (CunstomDataset): Child of CunstomDataset which can get image and annotations by index.
+       mosaic_pipeline (pipeline): Augmentations after mosaic
+       pipeline (pipeline): Augmentations after mosaic and mixup.
+       img_scale (Sequence[int]): image size after mosaic pipeline
+       enable_mosaic (bool): enable/disable mosiac aug, controlled by yolox process hook
+       mosaic_scale (Sequence[float]): center range of mosaic output
+       enable_mixup (bool): enable/disable mixup aug, controlled by yolox process hook
+       mixup_scale (Sequence[float]): image scale factor of mixup
+       pad_value (int): pad value
+
+   """
+    def __init__(self,
+                 dataset,
+                 pipeline,
+                 mosaic_pipeline=None,
+                 mixup_pipeline=None,
+                 img_scale=(640, 640),
+                 enable_mosaic=True,
+                 mosaic_scale=(0.5, 1.5),
+                 enable_mixup=True,
+                 mixup_scale=(0.5, 1.5),
+                 mixup_flip_ratio=0.5,
+                 pad_value=114):
+        self.dataset = dataset
+        self.CLASSES = dataset.CLASSES
+        if hasattr(self.dataset, 'flag'):
+            self.flag = dataset.flag
+        self.mosaic_pipeline = Compose(mosaic_pipeline) if mosaic_pipeline is not None else None
+        self.mixup_pipeline = Compose(mixup_pipeline) if mixup_pipeline is not None else None
+        self.pipeline = Compose(pipeline)
+        self.enable_mosaic = enable_mosaic
```

Comment:
It is recommended to add an interface for modifying the value, directly change the value of an attribute from outside is non-compliant.
Same comments for other attributes which would be modified by hook